Lib/smtplib.py

```diff
@@ -935,6 +935,10 @@ def send_message(self, msg, from_addr=None, to_addrs=None,
             header_prefix = 'Resent-'
         else:
             raise ValueError("message has more than one 'Resent-' header block")
+
+        # RFC 5322 section 3.6, 4th Paragraph
+        if msg.get('Date', None) is None:
+            msg['Date'] = email.utils.formatdate()
```

Comment:
According to RFC 5322 section 3.3:
> The date and time-of-day SHOULD express local time.

```suggestion
            msg['Date'] = email.utils.formatdate(localtime=True)
```

Reply:
nice catch!

Reply:
Since this is not enforced ("SHOULD"), we may want to add a parameter to allow format date to UTC. Let's see what do core developers think about that! :slightly_smiling_face:

Reply:
yup let's see what they say, my personal feelings is that it should "SHOULD" is pretty strong language and it would be better not to add an optional arg.

it's also making me think that I should rewrite the test, I'm not a super fan of patching `patch("email.utils.formatdate")` as it nows take away this "local" behaviour and prevents asserting on it

Reply:
It makes sense. Date can still be set manually before calling `send_message`.

Reply:
Yes, i wouldn't make it timelocal, but i don't harm to do it either.  This is  useful for knowing when the sender sent the message (understanding the context of the email). Like in Outlook, when your coworker is in a different timezone.